### PR TITLE
feat: kafka security (SASL)

### PR DESCRIPTION
**Pull request by shawnsarwar**
These changes add ACLs on a namespaced basis to Kafka topics. 
 - A broker level user is created and provided to all Kafka / ZK instances
 - A Kafka super user is created that can write to all topics.
 - The producer as a Kafka Superuser writes to topics like `{tenant}.{topicname}`
 - Each tenant has a kafka user named {tenant} created when the tenant is.
 - Each tenant has permissions on topics in their namespace {tenant}.*

The tenant password can be shared with the tenant, or derived for any tenant by any application using the `KAFKA_SECRET` environment variable.

### CODE REVIEW

File: docker-compose-base.yml [hunk 1]
```diff
@@ -216,23 +216,51 @@ services:
     environment:
       ZOOKEEPER_CLIENT_PORT: 32181
       ZOOKEEPER_TICK_TIME: 2000
+      ZOOKEEPER_INIT_LIMIT: 5
+      ZOOKEEPER_SYNC_LIMIT: 2
+      ZOOKEEPER_SESSION_TIMEOUT_MS: 750  # may be too short for prod.
```
Comment (adamvert) [inline docker-compose-base.yml]: Could you suggest a better value for prod? Also, could you add some comments for the other options - or do we have our preferred values documented somewhere?

File: kafka/make_credentials.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+source .env
+
+function gen_kafka_creds {
+    cat << EOF
+KafkaServer {
+   org.apache.kafka.common.security.scram.ScramLoginModule required;
```
Comment (adamvert) [inline kafka/make_credentials.sh]: Are these semicolons correct? If so, that's just plain weird 😕

Comment (shawnsarwar) [reply kafka/make_credentials.sh]: I agree, but they are. They separate sections (Scram from Plain) in this case.

File: docker-compose-base.yml (hunk 1 above)
Comment (shawnsarwar) [reply docker-compose-base.yml]: SYNC and INIT are defaults so they could be removed. 

Session timeout would depend on latency of the production network. On local, it's about 0.5ms. 750 is likely a fine value in prod, but there's no way to know at this point. If it's too short, ZK will close Kafka session when it shouldn't, which is suboptimal but not problematic. Having it set too long causes problems if Kafka goes down, restart and then ZK expects it to still be taking part in the old session.

**Comment by shawnsarwar** [comment]
Nope, looks good!

**Pull request merged**